google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/InstanceAdminRequests.java

```diff
@@ -0,0 +1,585 @@
+            com.google.bigtable.admin.v2.AppProfile.SingleClusterRouting.newBuilder()
+                .setClusterId(clusterId)
+                .setAllowTransactionalWrites(ignoreWarnings));
+        protoUpdateProfile.clearMultiClusterRoutingUseAny();
+        replacefields.add(SINGLE_CLUSTER__FIELDMASK);
+        return this;
+      }
+
+      @InternalApi
+      public com.google.bigtable.admin.v2.AppProfile toProto() {
+        return protoUpdateProfile.build();
+      }
+
+      @InternalApi
+      public FieldMask getPartialUpdateFieldMask() {
+        return FieldMask.newBuilder().addAllPaths(replacefields).build();
+      }
+    }
+
+    public interface RoutingPolicy {
+      RoutingPolicyCase name();
+    }
+
+    public static class RoutingPolicyNotSet implements RoutingPolicy {
+      private RoutingPolicyNotSet() {}
+
+      @Override
+      public RoutingPolicyCase name() {
+        return RoutingPolicyCase.ROUTINGPOLICY_NOT_SET;
+      }
+
+      @Override
+      public String toString() {
+        return MoreObjects.toStringHelper(this).add("name", name()).toString();
+      }
+    }
+
+    public static class MultiClusterRoutingUseAny implements RoutingPolicy {
+      private MultiClusterRoutingUseAny() {}
+
+      @Override
+      public RoutingPolicyCase name() {
+        return RoutingPolicyCase.MULTI_CLUSTER_ROUTING_USE_ANY;
+      }
+
+      @Override
+      public String toString() {
+        return MoreObjects.toStringHelper(this).add("name", name()).toString();
+      }
+    }
+
+    public static class SingleClusterRouting implements RoutingPolicy {
+      private final String clusterId;
+      private final boolean allowTransactionalWrites;
+
+      private SingleClusterRouting(String clusterId, boolean allowTransactionalWrites) {
+        this.clusterId = clusterId;
+        this.allowTransactionalWrites = allowTransactionalWrites;
+      }
+
+      private SingleClusterRouting(
+          com.google.bigtable.admin.v2.AppProfile.SingleClusterRouting routing) {
+        this.clusterId = routing.getClusterId();
+        this.allowTransactionalWrites = routing.getAllowTransactionalWrites();
+      }
+
+      public String getClusterId() {
+        return clusterId;
+      }
+
+      public boolean isAllowTransactionalWrites() {
+        return allowTransactionalWrites;
+      }
+
+      @Override
+      public RoutingPolicyCase name() {
+        return RoutingPolicyCase.SINGLE_CLUSTER_ROUTING;
+      }
+
+      @Override
+      public String toString() {
+        return MoreObjects.toStringHelper(this)
+            .add("name", name())
+            .add("clusterId", getClusterId())
+            .add("allowTransactionalWrites", isAllowTransactionalWrites())
+            .toString();
+      }
+    }
+  }
+
+  public static final class Policy {
+    private com.google.iam.v1.Policy.Builder protoPolicy = com.google.iam.v1.Policy.newBuilder();
+
+    public static Policy of(int version, Role role, List<String> members) {
```

Comment:
- version should be internal/hardcoded
- policies can be empty, so don't required role/members for construction